### ISSUE TITLE
Invalid moments shouldn't return valid Date objects

### ISSUE THREAD

**Issue by adrianrfreedman** [moment/moment#3430]
**Description of the Issue and Steps to Reproduce:**

Subtract does not seem to work when I create a date with an "overflow". See the output from the console below:
```js
> moment([2015,12]).toDate();
< Fri Jan 01 2016 00:00:00 GMT+1100 (AUS Eastern Daylight Time)
> moment([2015,12]).subtract(1,'days').toDate();
< Fri Jan 01 2016 00:00:00 GMT+1100 (AUS Eastern Daylight Time)
> moment([2015,12]).subtract(1,'day').toDate();
< Fri Jan 01 2016 00:00:00 GMT+1100 (AUS Eastern Daylight Time)
> moment([2015,12]).subtract(2,'day').toDate();
< Fri Jan 01 2016 00:00:00 GMT+1100 (AUS Eastern Daylight Time)
> moment([2015,11]).subtract(1,'day').toDate();
< Mon Nov 30 2015 00:00:00 GMT+1100 (AUS Eastern Daylight Time)
> moment([2016,0]).subtract(1,'day').toDate();
< Thu Dec 31 2015 00:00:00 GMT+1100 (AUS Eastern Daylight Time)
```

**Environment:**

Chrome Version 53.0.2785.101 m on Windows 7, Moment version 2.11.0
Also using DateRangePicker which works on top of jQuery UI but I get the same output irrespective of whether I include these libraries or not.

If you are reporting an issue, please run the following code in the environment you are using and include the output:
```js
console.log( (new Date()).toString())
console.log((new Date()).toLocaleString())
console.log( (new Date()).getTimezoneOffset())
console.log( navigator.userAgent)
console.log(moment.version)

daterangepicker.html:10 Tue Sep 13 2016 11:46:14 GMT+1000 (AUS Eastern Standard Time)
daterangepicker.html:11 13/09/2016, 11:46:14
daterangepicker.html:12 -600
daterangepicker.html:13 Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.101 Safari/537.36
daterangepicker.html:14 2.11.0
```

**Comment by maggiepint** [comment]
This is because we don't do overflow when passing an array to the moment constructor:

```js
moment([2015,12]).format()
"Invalid date"
```

As such, subtract is doing nothing.

It is a little weird that the date object held internally doesn't become invalid date. If there is a bug here, it would be that. @icambron or @ichernev is there a historical reason for this behavior that I'm not aware of?

**Comment by icambron** [comment]
We probably just never though to do it, TBH. IMO, `toDate` should return an invalid date from invalid moments.

**Comment by mj1856** [comment]
Besides `.toDate()`, also `.valueOf()` and `.unix()` return values on invalid moments.  Seems like the easiest thing to do would be to set `_d` to `new Date(NaN)` during construction of an invalid moment.

**Comment by samuelfullerthomas** [comment]
I've addressed this in #3506 if anyone wants to take a look at it.

**Comment by maggiepint** [comment]
Closed in favor of #3506
